src/datadog/trace_sampler_config.cpp

```diff
@@ -154,18 +53,7 @@ Expected<FinalizedTraceSamplerConfig> finalize_config(
     result.rules.push_back(std::move(catch_all));
   }
 
-  auto max_per_second = config.max_per_second;
-  if (auto limit_env = lookup(environment::DD_TRACE_RATE_LIMIT)) {
-    auto maybe_max_per_second = parse_double(*limit_env);
-    if (auto *error = maybe_max_per_second.if_error()) {
-      std::string prefix;
-      prefix += "While parsing ";
-      append(prefix, name(environment::DD_TRACE_RATE_LIMIT));
-      prefix += ": ";
-      return error->with_prefix(prefix);
-    }
-    max_per_second = *maybe_max_per_second;
-  }
+  auto max_per_second = config.max_per_second.value_or(200);
```

Comment:
Have to admit, that's weird. Two options:
- Complex data type provide their own `finalize` free function. They should own the default value to set and also load env conf.
- `finalize_conf` is the mastermind, set everything and default value. sub `finalize` free function always trust the input.
- `TracerConfig` is flat like earth. Make sense for `DatadogConfig` because 3/4 of it has nothing datadog related.

Reply:
Good points, not an easy decision. I will think about it and comment with my thoughts.